### PR TITLE
Plaintext connection to https endpoint hangs on jdk11: reproducer

### PR DESCRIPTION
**Pull request by raboof**
Refs #2614

## References

References #xxxx

### CODE REVIEW

**Comment by jrudolph** [comment]
@raboof, anything we could to about that?

**Comment by raboof** [comment]
I don't think I've looked into this before apart from recording the reproducer - I bet we can once we look closer ;)

File: akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala [hunk 1]
```diff
@@ -751,6 +752,30 @@ Host: example.com
 
       Await.result(binding.unbind(), 10.seconds)
     }
+
+    "produce a useful error message when connecting to a HTTPS endpoint over HTTP" in Utils.assertAllStagesStopped {
+      // akka/akka-http#1219
```
Comment (ennru) [inline akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala:757]: Isn't this about https://github.com/akka/akka-http/issues/2614 ?

**Comment by raboof** [comment]
> Why doesn't PR validation fail on this test? Aren't the tests run on JDK 11?

I think PR validation is on jdk8 (and we have jdk11 nightlies)

File: akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala (hunk 1 above)
Comment (raboof) [reply akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala:757]: Not sure what I was thinking
```suggestion
      // akka/akka-http#2614
```

**Comment by jrudolph** [comment]
Tried that with the fixes in https://github.com/akka/akka/pull/29121. It seems while it just hangs in the current version, with the fix, it will close the connection properly. The request will then fail with "The http server closed the connection unexpectedly before delivering responses for 1 outstanding requests" which is already better than hanging.

It seems for these kinds of errors TLSActor currently calls `fail(ex, closeTransport = false)` which means only cancellation is sent but no error. Not sure why that's the case.

**Comment by jrudolph** [comment]
> It seems for these kinds of errors TLSActor currently calls `fail(ex, closeTransport = false)` which means only cancellation is sent but no error. Not sure why that's the case.

Removing that makes `TlsSpec` "emit an error if the TLS handshake fails certificate checks" because it seems server and client might fail at the same time and there might be a race about which error is received first?

**Comment by raboof** [comment]
This seems to succeed now

**Comment by jrudolph** [comment]
Did you test on JDK 11? Would we expect it to succeed with the fix not yet released on 2.5.x?

**Comment by raboof** [comment]
Oops, indeed I was still on JDK8 here :confused: sorry about the noise.

**Comment by jrudolph** [comment]
Can now be restested with an update to 2.5.32.

**Comment by ignasi35** [comment]
Cant' test this locally due to #3556

**Comment by jrudolph** [comment]
Rebased on master where on JDK 11 it doesn't fail anymore with a timeout but now with the wrong exception because it seems the connection reset is received as a regular connection closure for some reason.

**Comment by jrudolph** [comment]
> Rebased on master where on JDK 11 it doesn't fail anymore with a timeout but now with the wrong exception because it seems the connection reset is received as a regular connection closure for some reason.

Seems to happen on both 2.5.32 and 2.6.10. Verified with wireshark that the server closes the connection regularly.

The culprit seems to still be the TLSActor which sends failures user-side but maybe not towards the network?

**Comment by raboof** [comment]
reported on the Akka side as https://github.com/akka/akka/issues/29957

**Pull request closed without merging**